cmd/rootMain.go

```diff
@@ -122,16 +117,77 @@ Try 'help'`)
 	)
 	comm.Dir = mountpath
 
+	// Inspired by https://blog.nelhage.com/2011/02/changing-ctty/. After the child shell has
+	// started, we re-parent the Wash daemon to be in a child session of the shell so that when Wash
+	// prompts for input it's within the TTY of that terminal.
+	if startErr := comm.Start(); startErr != nil {
+		cmdutil.ErrPrintf("%v\n", startErr)
+		return exitCode{1}
+	}
+
+	codeCh := make(chan int)
+	go func() {
+		if runErr := comm.Wait(); runErr != nil {
+			if exitErr, ok := runErr.(*exec.ExitError); ok {
+				codeCh <- exitErr.ExitCode()
+			} else {
+				cmdutil.ErrPrintf("%v\n", runErr)
+				codeCh <- 1
+			}
+		} else {
+			codeCh <- 0
+		}
+		close(codeCh)
+	}()
+
 	var exit exitCode
-	if runErr := comm.Run(); runErr != nil {
-		if exitErr, ok := runErr.(*exec.ExitError); ok {
-			exit.value = exitErr.ExitCode()
+	killCommand := func() exitCode {
```

Comment:
Nit: `killCommand` => `killShellProcess` ? Makes it a bit easier to read.

Reply:
Sure.